src/set_theory/ordinal_arithmetic.lean

```diff
@@ -2020,22 +2020,31 @@ end
 theorem CNF_pairwise (b := omega) (o) : (CNF b o).pairwise (λ p q, prod.fst q < p.1) :=
 (CNF_pairwise_aux _ _).2
 
-theorem CNF_fst_le_log (b := omega) (o) : ∀ p ∈ CNF b o, prod.fst p ≤ log b o :=
-(CNF_pairwise_aux _ _).1
+theorem CNF_fst_le_log (b := omega) {o p} : p ∈ CNF b o → prod.fst p ≤ log b o :=
+(CNF_pairwise_aux _ _).1 p
 
-theorem CNF_fst_le (b := omega) (o) (p ∈ CNF b o) : prod.fst p ≤ o :=
-le_trans (CNF_fst_le_log _ _ p H) (log_le_self _ _)
+theorem CNF_fst_le (b := omega) {o p} (hp : p ∈ CNF b o) : prod.fst p ≤ o :=
+(CNF_fst_le_log _ hp).trans (log_le_self _ _)
 
-theorem CNF_snd_lt {b : ordinal} (b1 : 1 < b) (o) : ∀ p ∈ CNF b o, prod.snd p < b :=
+/-- This theorem exists to factor out commonalities between the proofs. -/
+private theorem CNF_snd_lt_aux {b : ordinal} (b1 : 1 < b) {o} :
+  ∀ p ∈ CNF b o, prod.snd p < b ∧ 0 < p.2 :=
```

Comment:
```suggestion
  ∀ p ∈ CNF b o, p.2 < b ∧ 0 < p.2 :=
```
maybe you can uniformly change the `prod.fst` and `prod.snd` explicitly to `.1` and `.2`? If that fails to elaborate, you can give the type of `p` explicitly.

Reply:
Sure! Doing this actually revealed a hidden problem, that a bunch of theorems had two universes for whatever reason. I've fixed accordingly.